client/src/controllers/ClipboardController.ts

```diff
@@ -0,0 +1,35 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class ClipboardController extends Controller<HTMLElement> {
+  static targets = ['value'];
+
+  declare valueTarget:
+    | HTMLInputElement
+    | HTMLTextAreaElement
+    | HTMLSelectElement;
+
+  copy() {
+    this.dispatch('start');
+    if (!navigator.clipboard) {
+      this.showError();
+      return;
+    }
+
+    const value = this.valueTarget.value;
+
+    if (!value) return;
+
+    navigator.clipboard.writeText(value).then(
+      () => this.showSuccess(),
+      () => this.showError(),
```

Comment:
It's probably more common to put the rejection handler inside `catch`, unless I am missing something specific we need to do here.
e.g.
```
   navigator.clipboard.writeText(value).then(
      () => this.showSuccess(),
  ).catch(() => this.showError())
```